object/datastore_file_manager.go

```diff
@@ -115,16 +116,21 @@ func (m *DatastoreFileManager) CopyFile(ctx context.Context, src string, dst str
 }
 
 // Copy dispatches to the appropriate FileManager or VirtualDiskManager Copy method based on file name extension
-func (m *DatastoreFileManager) Copy(ctx context.Context, src string, dst string) error {
+func (m *DatastoreFileManager) Copy(ctx context.Context, src string, dst string, spec ...types.BaseVirtualDiskSpec) error {
 	srcp := m.Path(src)
 	dstp := m.Path(dst)
 
 	f := m.FileManager.CopyDatastoreFile
 
 	if srcp.IsVMDK() {
+		var destSpec types.BaseVirtualDiskSpec
+		if len(spec) != 0 {
+			destSpec = spec[0]
+		}
+
 		// types.VirtualDiskSpec=nil as it is not implemented by vCenter
 		f = func(ctx context.Context, src string, srcDC *Datacenter, dst string, dstDC *Datacenter, force bool) (*Task, error) {
-			return m.VirtualDiskManager.CopyVirtualDisk(ctx, src, srcDC, dst, dstDC, nil, force)
+			return m.VirtualDiskManager.CopyVirtualDisk(ctx, src, srcDC, dst, dstDC, destSpec, force)
```

Comment:
Would it be better to use `dstSpec` for uniformity?

Reply:
Yes, thanks, I'll fix that